fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/MatchCandidate.java

```diff
@@ -290,6 +299,18 @@ static Iterable<MatchCandidate> fromIndexDefinition(@Nonnull final RecordMetaDat
                     .ifPresent(resultBuilder::add);
         }
 
+        if (AggregateIndexExpansionVisitor.isAggregateIndex(type, index.getRootExpression())) {
```

Comment:
It seems like we could actually make all of this a `switch`?

Reply:
done.